cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -327,6 +330,15 @@ public void performAction(HttpRequest request, HttpResponder responder,
                             @PathParam("type") String type,
                             @PathParam("id") String id,
                             @PathParam("action") String action) {
+    // If the app is deployed and it is an Application Template, then don't allow any action.
+    // Operations are only allowed through Adapter Lifecycle management.
+    if (store.getApplication(Id.Application.from(namespaceId, appId)) != null &&
```

Comment:
looks like we check on delete, but not on create. should check on create as well.